crates/services/relayer/src/service/get_logs.rs

```diff
@@ -57,16 +58,14 @@ where
                         let oldest_block = page.oldest();
                         let latest_block = page.latest();
 
-                        // Reduce the page.
-                        let page = page.reduce();
-
                         // Get the logs and return the reduced page.
                         eth_node
                             .get_logs(&filter)
                             .await
                             .map_err(|err| {
                                 let ProviderError::JsonRpcClientError(err) = err else {
-                                    return err
+                                    page_sizer.update(0, true);
```

Comment:
```suggestion
                                    // we optimistically reduce the page size for any rpc error, to reduce the work to check if it is actually because of "too many logs"
                                    page_sizer.update(0, true);
```